sympy/geometry/ellipse.py

```diff
@@ -327,16 +328,12 @@ def circumference(self):
         >>> p1 = Point(0, 0)
         >>> e1 = Ellipse(p1, 3, 1)
         >>> e1.circumference
-        12*Integral(sqrt((-8*_x**2/9 + 1)/(-_x**2 + 1)), (_x, 0, 1))
+        4*elliptic_e(2*sqrt(2)/3)
 
         """
-        from sympy import Integral
         if self.eccentricity == 1:
-            return 2*pi*self.hradius
-        else:
-            x = Dummy('x', real=True)
-            return 4*self.major*Integral(
-                sqrt((1 - (self.eccentricity*x)**2)/(1 - x**2)), (x, 0, 1))
+            return 2 * pi * self.hradius
+        return 4 * self.minor * elliptic_e(self.eccentricity)
```

Comment:
I think this should be `self.major`. If the ellipse is very flat, `self.minor` is near 0, and elliptic_e term is close to 1. But the perimeter in this case is about `4*major axis`. [Wikipedia agrees](https://en.wikipedia.org/wiki/Ellipse#Circumference)